### PR TITLE
Upgrade from node16 to node20

### PR DESCRIPTION
**Pull request by Nishnha**
Node16 is now EOL. The next LTS release is node18 which has support until 2025, but node20 has support until 2026.

https://nodejs.dev/en/about/releases/

### CODE REVIEW

**Comment by Nishnha** [comment]
CI is currently failing, but the tests all passed. One of our tests generates an error and it seems like the Actions runner is picking up on that and reporting the step as failed?

> > dependabot-pull-request-action@1.6.0 test
> > jest
> 
> PASS src/main.test.ts
> Error: github-token is not set! Please add 'github-token: "${{ secrets.GITHUB_TOKEN }}"' to your workflow file.





PASS src/dependabot/update_metadata.test.ts
PASS src/dependabot/verified_commits.test.ts
PASS src/dependabot/output.test.ts

Test Suites: 4 passed, 4 total
Tests:       40 passed, 40 total
Snapshots:   0 total
Time:        4.20[6](https://github.com/dependabot/fetch-metadata/actions/runs/6266312498/job/17017062301?pr=443#step:6:7) s
Ran all test suites.

**Comment by jeffwidman** [comment]
Is the underlying actions issue mentioned here related to the CI failures?
* https://github.com/dependabot/fetch-metadata/issues/335

**Comment by Nishnha** [comment]
> Is the underlying actions issue mentioned here related to the CI failures?
> 
> * [Migrate to node 18 whenever Actions adds support for it. #335](https://github.com/dependabot/fetch-metadata/issues/335)

The original issue was but I think node 20 has shipped on Actions runners. Now I see a new error around a missing GitHub token that I can take care of in a bit.

**Comment by ParanoidUser** [comment]
As we near the deadline in Spring 2024, this action has begun generating deprecation warnings. Are there any obstacles in the migration process? I anticipate this becoming a prominent topic soon, as more individuals express frustration with these warnings.
```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: dependabot/fetch-metadata@c9c4182bf1b97f5224aee3906fd373f6b61b4526. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

**Comment by Nishnha** [comment]
> Are there any obstacles in the migration process?

Nope just got busy. We're going to release this as a major version update since it breaks compatibility with older GitHub Enterprise Server versions that don't ship with Node20. I'll create a minor release for whatever changes are on main first, though.

**Comment by hanseartic** [comment]
Would really appreciate a release /w node20 support soon. Most of the actions are already migrated....

**Comment by jeffwidman** [comment]
@Nishnha I'm unclear what's the next step here? 

Do we need a minor release to catchup to `HEAD` and then merge this and cut a major release?

Should we land any of the other outstanding PR's first? I can take a look and make judgement calls but assume you've already looked at this?

File: package.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
     "approve"
   ],
   "engines": {
-    "node": ">=16.0.0",
-    "npm": ">=7.10.0"
+    "node": ">=20.0.0",
+    "npm": ">=10.0.0"
```
[the remark this answers is not in the document]
Comment (jeffwidman) [reply package.json]: Thanks @hanseartic

**Review by jeffwidman — APPROVED** [review]

**Pull request merged**